edi_sale_oca/components/process.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2021 Camptocamp SA
+# @author: Simone Orsi <simone.orsi@camptocamp.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import _
+from odoo.exceptions import UserError
+
+from odoo.addons.component.core import Component
+
+
+class EDIExchangeSOInput(Component):
+    """Process sale orders."""
+
+    _name = "edi.input.sale.order.process"
+    _inherit = "edi.component.input.mixin"
+    _usage = "input.process.sale.order"
+
+    def __init__(self, work_context):
+        super().__init__(work_context)
+        self.settings = {}
+        # Suppor legacy key `self.type_settings`
+        for key in ("sale_order", "sale_order_import"):
```

Comment:
IMO we can remove all these special keys. Including `wiz_ctx`.

If fixed a long time ago the propagation of `env_ctx` down the edi stack.
This means that, in theory, we can pass all configurations via `env_ctx`.

IMO this has a bunch of benefits:

1. contextualize the conf inside the specific component's settings
2. eliminate superfluous code

We could even keep the `wiz_ctx` inside the `env_ctx` but I wonder if it makes sense. 
We could get rid of line 56 and simply have something like:

```
components:
    process:
        usage: input.process.sale.order
        env_ctx:
            # values for the wizard
            default_confirm_order: true
            default_price_source: order
            # custom keys, whatever you need
            random_one: true
```

I'm sorry because I should have written this stuff in the roadmap but is better to do it now before we start using it otherwise we'll have to support the old style conf for another bunch of versions.
If you can go ahead w/ this, pls do it in its own commit to be able to cherry-pick it.

Reply:
I updated, could you have a look?